livekit/src/room/participant/mod.rs

```diff
@@ -248,8 +248,19 @@ pub(super) fn add_publication(
     publication.on_muted({
         let events = inner.events.clone();
         let participant = participant.clone();
+        let rtc_engine = inner.rtc_engine.clone();
         move |publication| {
             if let Some(cb) = events.track_muted.lock().as_ref() {
+                let rtc_engine = rtc_engine.clone();
```

Comment:
When unmuting or muting again, there is no guarantee that these tasks will run in order. Seems like a race could happen here?

Reply:
as in the `mute_track().await` could take longer then the unmute after it?

Reply:
not necessarily longer, but the line could be executed before

Reply:
new commit locks mute and unmute under the same mutex

Reply:
I saw you added the mutex, but this isn't fixing the issue. The race could still be there